examples/ErrorHandling/HandleKeywordPolicyViolations.php

```diff
@@ -0,0 +1,263 @@
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID without hyphens
+     * @param int $adGroupId the ad group ID to add a keyword to
+     * @param string $keywordText the keyword text to add
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId,
+        string $keywordText
+    ) {
+        // Configures the keyword text and match type settings.
+        $keywordInfo = new KeywordInfo([
+            'text' => new StringValue(['value' => $keywordText]),
+            'match_type' => KeywordMatchType::EXACT
+        ]);
+
+        // Constructs an ad group criterion using the keyword text info above.
+        $adGroupCriterion = new AdGroupCriterion([
+            'ad_group' => new StringValue(
+                ['value' => ResourceNames::forAdGroup($customerId, $adGroupId)]
+            ),
+            'status' => AdGroupCriterionStatus::ENABLED,
+            'keyword' => $keywordInfo
+        ]);
+
+        $adGroupCriterionOperation = new AdGroupCriterionOperation();
+        $adGroupCriterionOperation->setCreate($adGroupCriterion);
+        $adGroupCriterionServiceClient = $googleAdsClient->getAdGroupCriterionServiceClient();
+
+        try {
+            // Try sending a mutate request to add the keyword.
+            $response = $adGroupCriterionServiceClient->mutateAdGroupCriteria(
+                $customerId,
+                [$adGroupCriterionOperation]
+            );
+            printf(
+                "Added a keyword with resource name '%s'.%s",
+                $response->getResults()[0]->getResourceName(),
+                PHP_EOL
+            );
+            // You may hit this line when you run this example using the default keyword for the
+            // second time. This is because the keyword will already be exempted and the mutate
+            // request can thus be successful.
+            return;
+        } catch (GoogleAdsException $googleAdsException) {
+            $exemptPolicyViolationKeys = self::fetchExemptPolicyViolationKeys($googleAdsException);
```

Comment:
I would add a comment to state that reaching this catch method is expected due to the violation policy.

Reply:
I think it'd not be necessary because
1) We already explain this at the file-level doc.
2) It's not expected always partly because of the cases that you point out (normal keyword).

I'd keep it concise here. The private method PHPDoc will do its task to explain things in there too.